src/Xamarin.CodeAnalysis/RequiresSuperAttributeCodeFixProvider.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+//
+// RequiresSuperAttributeCodeFixProvider.cs:
+//
+// Authors:
+//   Madeline Zhang  <v-madzh@microsoft.com>
+//
+// Copyright 2020 Xamarin Inc.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+
+namespace RequiresSuperAttribute
+{
+	[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (RequiresSuperAnalyzerCodeFixProvider)), Shared]
+	public class RequiresSuperAnalyzerCodeFixProvider : CodeFixProvider
+	{
+		const string Title = "Add base method call";
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds {
+			get { return ImmutableArray.Create (RequiresSuperAttributeAnalyzer.DiagnosticId); }
+		}
+
+		public sealed override FixAllProvider GetFixAllProvider ()
+		{
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+			var diagnostic = context.Diagnostics.First ();
+			var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+			// Find the method declaration identified by the diagnostic.
+			var methodDeclaration =
+			  root.FindToken (diagnosticSpan.Start).Parent.AncestorsAndSelf ()
+			  .OfType<MethodDeclarationSyntax> ().First ();
+
+			// Register a code action that will invoke the fix.
+			context.RegisterCodeFix (CodeAction.Create (Title, c => FixSuperAsync (context.Document, methodDeclaration, c), equivalenceKey: Title), diagnostic);
+		}
+
+		private async Task<Document> FixSuperAsync (Document document,
+		  MethodDeclarationSyntax methodDeclaration,
+		  CancellationToken cancellationToken)
+		{
+			//Find the node to fix.
+			var semanticModel = await document.GetSemanticModelAsync (cancellationToken);
+			var memberSymbol = semanticModel.GetDeclaredSymbol (methodDeclaration) as IMethodSymbol;
```

Comment:
Using an as but AFAIK not checking if is null. This will return in an issuing in line 67.